content/blog/create-a-general-purpose-kubeconfig-file-in-hpe-greenlake-for-containers.md

```diff
@@ -0,0 +1,261 @@
+
+
+
+![](/img/hpe-ecp-dashboard.png "HPE ECP Dashboard")
+
+The kubeconfig file that can be downloaded from the HPE Ezmeral Runtime Enterprise dashboard comes up with a couple of issues:
+
+1. The kubeconfig file is tied to the user who logs in to HPE GreenLake for Containers. There are many use cases that use simple scripts, such as in Bash, that call out to the kubectl or a proper client library to access the Kubernetes cluster with the kubeconfig file outside the cluster. They are not tied to any particular user. Providing a kubeconfig file that’s tied to your user is not considered to be a clean design due to the fact that each user may have different privileges, and providing the kubeconfig file with ability to access to the cluster may violate the [Principle of Least Privilege](https://en.wikipedia.org/wiki/Principle_of_least_privilege). 
+
+2. Since launching from HPE GreenLake Central to the HPE Ezmeral Runtime Enterprise is configured through SAML SSO, a session token is fetched and added to the kubeconfig file each time you launch the HPE Ezmeral Runtime Enterprise dashboard. With HPE GreenLake for Containers, the session token is configured to expire after an hour. You will be unable to use the downloaded kubeconfig file to access the cluster after the token expires. You will have to relaunch the dashboard and once again download the kubeconfig file.
+
+3. The kubeconfig file is generated to include some specific commands to show version, authenticate, and refresh the HPE Ezmeral Runtime Enterprise environment. The standard kubectl tool installed from the [official Kubernetes site](https://kubernetes.io/docs/tasks/tools/) does not work with this kubeconfig file. You have to download the `HPE kubectl plugin`, available from the same dashboard, and use it together with the kubectl and the kubeconfig file. This causes issue due to the fact that some services and tools, such as `Azure DevOps` and `Jenkins`, use the standard kubectl to access and deploy applications to the Kubernetes clusters. 
```

Comment:
In the last sentence, please modify "This causes issue due to..." to either "This causes an issue due to..." or "This causes issues due to..."